### PR TITLE
Tests for set D3 State (1/2)

### PR DESCRIPTION
**Pull request by jdnordy**
### **CHANGES**
Refactored setGlobalVars --> setD3State.
modularized setD3State by creating helper functions.
Turned setD3State into a pure function for better testing.

### **TESTS**
Implemented tests for all helper functions.

### CODE REVIEW

**Review by michaelbayday — CHANGES_REQUESTED** [review]
Nice refactor of the code man!

File: tsconfig.json [hunk 1]
```diff
@@ -13,7 +13,11 @@
     "dist/**/*",
     "release/**/*",
     "webpack.main.ext.js",
+<<<<<<< HEAD
```
Comment (michaelbayday) [inline tsconfig.json]: can you clean up this merge conflict

Comment (jdnordy) [reply tsconfig.json]: just pushed it. I removed tests from exclude so we still get type checking in our tests

**Review by michaelbayday — APPROVED** [review]
nice work my friend!

**Pull request merged**